frontend/aml_dashboard/src/index.js

```diff
@@ -0,0 +1,746 @@
+
+const model_received = text('No model received');
+model_received.title = 'AML Model';
+
+const collaborative_status = text('Not received');
+collaborative_status.title = 'AML Collaborative Learning Status';
+
+const search_statistics = button('Search for statistics');
+search_statistics.title = 'AML Statistics Fetcher'
+
+search_statistics.$click.subscribe( async () => {
+  console.log('Search statistics');
+  collaborative_status.$value.set(' <h1>Searching statistics...</h1> <br> <img src="https://i.gifer.com/Cad.gif"> ');
+
+  fetch(url_fetcher + 'statistics', {
+    method: "GET", // *GET, POST, PUT, DELETE, etc.
+    mode: "cors", // no-cors, *cors, same-origin
+    cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
+    credentials: "omit", // include, *same-origin, omit
+    headers: {
+      "Content-Type": "application/json",
+    },
+    redirect: "follow", // manual, *follow, error
+    referrerPolicy: "no-referrer", // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+  })
+  .then(response => response.json())
+  .then(async json => {
+    console.log(json);
+    console.log('STATISTICS RECEIVED');
+
+    const name = json.name; // Extracting the value of 'name' field from the JSON response
+
+    collaborative_status.$value.set('<h1>Statistics received !</h1> ');
+    statistics_received.$value.set('<h2>Statistics received: ' + name + '</h2>');
+
+  });
+
+});
+
+const request_model = button('Request model');
+request_model.title = 'AML Model Fetcher'
+
+request_model.$click.subscribe( async () => {
+
+  if (statistics_received.$value.get() == 'No statistics received') {
+    throwError(new Error('No statistics have been received'));
+  } else {
+
+    console.log('Request model');
+    collaborative_status.$value.set(' <h1>Requesting model...</h1> <br> <img src="https://i.gifer.com/Cad.gif">');
+
+    const v = await trainingSet.find();
+
+    // TODO: save the training set to a file each time is updated
+    save_to_file(v, "training_set_.json"); // Save the training set to a file
+
+    fetch(url_fetcher + 'model', {
+      method: "GET", // *GET, POST, PUT, DELETE, etc.
+      mode: "cors", // no-cors, *cors, same-origin
+      cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
+      credentials: "omit", // include, *same-origin, omit
+      headers: {
+        "Content-Type": "application/json",
+      },
+      redirect: "follow", // manual, *follow, error
+      referrerPolicy: "no-referrer", // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+    })
+    .then(response => response.json())
+    .then(async json => {
+      console.log(json);
+      console.log('MODEL RECEIVED');
+
+      save_to_file(json, "model_.json");
+
+      aml_model['ready'] = true;
+
```

Comment:
Please take into account the case where no model is found.


```suggestion

      collaborative_status.$value.set('<h1>Model not found</h1> ');
      model_received.$value.set('<h2>Model not found</h2>');
```

Reply:
Addressed in #2